scripts/scil_denoise_sf.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to compute per-vertices hemisphere-aware (asymmetric) Gaussian
+filtering of spherical functions (SF) given an array of spherical harmonics
+(SH) coefficients.
+
+The resulting SF can be expressed using a full SH basis (to keep the
+asymmetry resulting from the filtering) or a symmetric SH basis (where the
+effect of the filtering is a simple denoising).
+"""
+
+import argparse
+import logging
+import os
+
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg, assert_inputs_exist,
+                             assert_outputs_exist, add_sh_basis_args)
+
+from scilpy.denoise.asym_enhancement import local_asym_gaussian_filtering
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('--out_dir', default='.',
+                   help='Output directory. Default is current directory.')
+
+    p.add_argument('--out_sh', default='out_filtered.nii.gz',
+                   help='File name for averaged signal. [%(default)s]')
+
+    p.add_argument('--out_mask', default='mask.nii.gz',
+                   help='File name for output mask. [%(default)s]')
+
+    p.add_argument('--mask_eps', default=1e-16,
+                   help='Threshold on SH coefficients norm for output mask.'
+                   ' [%(default)s]')
+
+    p.add_argument(
+        '--sh_order', default=8, type=int,
+        help='SH order of the input. [%(default)s]')
+
+    p.add_argument(
+        '--sphere', default='repulsion724', type=str,
+        help='Sphere used for the SH projection. [%(default)s]'
+    )
+
+    p.add_argument(
```

Comment:
Same below, I prefer having at least the name and default directly on the same line. Easier to read, and end parathesis on the same line:

    p.add_argument( '--sigma', default=1.0,
                                type=float,
                                help='Sigma of the gaussian to use. [%(default)s]')